ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -789,6 +824,8 @@ wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
       Aws::MakeShared<S3LogSystem>("storage", s3->wtApi, s3->verbose));
     Aws::InitAPI(options);
 
+    s3->log = new S3LogSystem(s3->wtApi, s3->verbose);
```

Comment:
What's the purpose of this allocation here? Aren't we already allocating a S3LogSystem on lines 823-824?

Reply:
I believe lines 823-824 are just integrating AWS logs with WiredTiger logs. This allocation is needed to call `LogVerboseMessage` to log custom messages in our extension error handling.

Reply:
This is actually double allocation, line 825, Aws::MakeShared allocates an instance of the logger. We have to use the same logger for both AWS and otherwise.